src/app/_lib/_types/types.ts

```diff
@@ -0,0 +1,57 @@
+import { z } from "zod";
+import { registerBodySchema } from "./schemas";
+
+export type FormInput = {
+  label: string;
+  type: string;
+  id: string;
+  placeholder: string;
+};
+
+export type FormData_t = {
+  title: string;
+  button: string;
+};
+
+export type RegisterInputs = {
+  email: string;
+  username: string;
+  password: string;
+};
+
+export type RegisterBody = z.infer<typeof registerBodySchema>;
+
+export type InputsEdited = {
+  email: false;
+  username: false;
+  password: false;
+};
+
+export type ServerResponse = {
+  success: boolean;
+  message: string;
+};
+
+export type User =
+  | ({
+      id: number;
+    } & RegisterBody)
+  | undefined;
```

Comment:
The Prettier extension is making this look alot worse than it is
Basically, this new `User `type is identical to the `RegisterBody `type, except it has an id. ( `{id: number} & RegisterBody` )